gcloud-java-dns/src/main/java/com/google/gcloud/dns/Zone.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.gcloud.Page;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+
+/**
+ * A Google Cloud DNS Zone object.
+ *
+ * <p>A zone is the container for all of your DNS records that share the same DNS name prefix, for
+ * example, example.com. Zones are automatically assigned a set of name servers when they are
+ * created to handle responding to DNS queries for that zone. A zone has quotas for the number of
+ * resource records that it can include.
+ *
+ * @see <a href="https://cloud.google.com/dns/zones/">Google Cloud DNS managed zone
+ * documentation</a>
+ */
+public class Zone implements Serializable {
+
+  // TODO(mderka) Zone and zoneInfo to be merged. Opened issue #605.
+
+  private static final long serialVersionUID = -4012581571095484813L;
+  private final ZoneInfo zoneInfo;
+  private final Dns dns;
+
+  /**
+   * Constructs a {@code Zone} object that contains the given {@code zoneInfo}.
+   */
+  public Zone(Dns dns, ZoneInfo zoneInfo) {
+    this.zoneInfo = checkNotNull(zoneInfo);
+    this.dns = checkNotNull(dns);
+  }
+
+  /**
+   * Constructs a {@code Zone} object that contains meta information received from the Google Cloud
```

Comment:
In noticed that also above we refer to the content of  `XXXInfo` objects as: metadata, information or meta. I think that only "information" should be clear enough.